sdk/extensions/Azure.Extensions.AspNetCore.Configuration.Secrets/tests/ConfigurationSecretsFunctionalTests.cs

```diff
@@ -33,6 +33,8 @@ public async Task SecretsAreLoadedFromKeyVault()
             Assert.AreEqual("2", configuration["TestSecret2"]);
             Assert.AreEqual("3", configuration["Nested:TestSecret3"]);
 
+            // KeyVault time resolution is 1sec we can't detect a change faster than that
```

Comment:
From the comments, it seems like we would want to add the delay after updating the value and before checking for the new value. So, between `SetSecretAsync()` and `Reload()`.
Or, do you want to say the client does not allow updates to the same key with less than 1 second of granularity?

Reply:
The KeyVault stores `updated` property on the secret with 1 sec fidelity. So if the key is updated twice during the same second we won't be able to detect the change. That's why the delay is between the first `SetSecretAsync` and the second `SetSecretAsync`.

Reply:
Ah, that's interesting. May have been better then for the API to allow clients to set `updated` so they can avoid this gotcha.

Reply:
I think KeyVault service knows the exact time but for some reason, they decided to only store it to 1-second precision.

Even if there was a way to pass the exact value they won't store it.